src/festim/material.py

```diff
@@ -76,19 +84,34 @@ def __init__(
         self.heat_capacity = heat_capacity
         self.name = name
         self.solubility_law = solubility_law
+        self.D = D
```

Comment:
Can we add a check that a user cannot provide both `D_0` and `D` at the same time?